fetcher_test.go

```diff
@@ -0,0 +1,102 @@
+package fetcher_test
```

Comment:
Perhaps an additional test for error handling too? Should the fetcher also handle panics in the `fetch`? 

“tests are affidavits”; in other words, the tests are ideally exhaustive of use cases for the library and introduce users to what’s possible

Reply:
done, feel free to review again. TLDR, panics will be caught, returned as an error of type `fetcher.PanicError`, and printing them includes the stack trace.